jwst/ramp_fitting/ramp_fit_step.py

```diff
@@ -456,6 +458,30 @@ def process(self, step_input):
 
             int_times = result.int_times
 
+            # Set the DO_NOT_USE bit in the groupdq values for groups before firstgroup
+            # and groups after lastgroup
+            firstgroup = self.firstgroup
+            lastgroup = self.lastgroup
+
+            if firstgroup is None:
+                firstgroup = 0
+            if lastgroup is None:
+                lastgroup = ngroups - 1
+            if firstgroup < 0:
+                log.warning("first group < 0, reset to 0")
+                firstgroup = 0
+            if lastgroup >= ngroups:
+                log.warning(f"Last group number >= #groups ({ngroups}), reset to {ngroups-1}")
+            if firstgroup >= lastgroup:
+                log.warning(f"firstgroup ({firstgroup}) cannot be >= lastgroup ({lastgroup})")
+                log.warning("Group selectors ignored")
+                firstgroup = 0
+                lastgroup = ngroups - 1
```

Comment:
It looks like these conditionals lack test coverage.  I would suggest using `with pytest.warns()` blocks to ensure the warning is emitted and to ensure that the behavior is as you expect.
https://docs.pytest.org/en/stable/reference/reference.html#pytest.warns

Reply:
I added a new test that just tests the warning emissions